### ISSUE TITLE
CBL 2.1.1 on MacOS never complete the sync

### ISSUE THREAD

**Issue by fedal-mas** [couchbase/couchbase-lite-ios#2208]
CBL isn't syncing all documents/revisions. In my case it get stuck forever saying that it has 27 changes to complete the sync.

Here is the [CBL log](https://gist.github.com/fedal-mas/48984b8d8b5a81e0b30ef8ea5ad37f69).

--------
* Version:     2.1.1
* Client OS: MacOS Mojave (10.14)
* Server:      CB Server 5.5.1 and SG 2.1.0

**Comment by fedal-mas** [comment]
@pasin, updated the CBL log and here is the [SG Log](https://github.com/couchbase/couchbase-lite-ios/files/2403058/SGLog.zip).

**Comment by pasin** [comment]
From CB Lite log, it seems like the pull replicator was waiting for SG to send the changes. From the SG log, I saw the errors like:

```
2018-09-20T17:35:40.351-03:00 [WRN] [c1f6fae] blipHandler can't get doc "-FAQCU0V_t8USQlmi_dKaOt"/2-6c2f99dc1f0e331116dbaa48fe2459c7: key not found -- rest.(*blipHandler).sendRevision() at blip_sync.go:546
```

I guessed those revisions are purged from SG/Couchbase Server. This seems like https://github.com/couchbase/sync_gateway/issues/3738 issue.

**Comment by pasin** [comment]
Closed the issue and let's track this issue from couchbase/sync_gateway#3738 instead.

**Comment by pasin** [comment]
I think I will let this issue opened until couchbase/sync_gateway#3738 gets resolved so that we can track it here as well.

**Comment by fedal-mas** [comment]
@pasin nice, I will keep an eye on the other issue. Thanks!

**Comment by pasin** [comment]
SG fix for the issue will be available in v2.1.1 and CBL completed progress will be available in v2.1.2. Thanks.
